src/main.rs

```diff
@@ -50,26 +51,20 @@ impl EventHandler for Handler {
             return;
         }
 
-        let re: Regex = Regex::new(r"^/?(r/).+").expect("Failed to initialise regex pattern");
-
-        if !re.is_match(&message.content) {
-            return;
-        }
+        let path = match content_to_path(&message.content) {
+            Some(path) => path,
+            None => return,
+        };
 
         message
             .delete(ctx.http.clone())
             .await
             .expect("Couldn't delete message");
 
-        let split_message: Vec<&str> = message.content.split('/').collect();
```

Comment:
Your original regex can contain a slash at the start - which if you split on it here can add an extra item to the front of the `Vec`